### PR TITLE
Update GitHub Actions workflow to trigger prod deploys and build version.json correctly

### PR DESCRIPTION
**Pull request by pattersonbl2**
prod changes:
adding new tag regex to prod 
removing immutable flag from build step
formatting fixes 
change fetch-depth to get all history 

stage changes: 
change fetch-depth to get all history

### CODE REVIEW

File: .github/workflows/gcp_deploy_prod.yaml [hunk 1]
```diff
@@ -50,7 +52,7 @@ jobs:
 
       - name: Build the website
         run: |
-          yarn install --immutable
+          yarn install
```
Comment (willdurand) [inline .github/workflows/gcp_deploy_prod.yaml:55]: Let's keep `--immutable` here and above

File: .github/workflows/gcp_deploy_prod.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ name: Deploy to GCS Prod
 on:
   push:
     tags:
-      - "20[0-9][0-9].[0-9][0-9].[0-9][0-9](-[0-9]+)?"
+      - "202[0-9].[0-9][0-9].[0-9][0-9]"
```
[the remark this answers is not in the document]
Comment (pattersonbl2) [reply .github/workflows/gcp_deploy_prod.yaml]: The previous way didn't trigger the action like i wanted but i was able to trigger it  by adding a * to the tag regex.  I have just pushed up this change.

File: .github/workflows/gcp_deploy_prod.yaml (hunk 1 above)
Comment (pattersonbl2) [reply .github/workflows/gcp_deploy_prod.yaml]: added both back.

File: .github/workflows/gcp_deploy_prod.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name: Deploy to GCS Prod
 on:
   push:
     tags:
-      - "20[0-9][0-9].[0-9][0-9].[0-9][0-9](-[0-9]+)?"
+      - "202[0-9].[0-9][0-9].[0-9][0-9]*"
```
Comment (willdurand) [inline .github/workflows/gcp_deploy_prod.yaml:6]: Maybe simplify to allow any tag that starts with `20`. We don't need to be very precise here, I think.

```suggestion
      - "20*"
```

File: .github/workflows/gcp_deploy_prod.yaml [hunk 4]
```diff
@@ -50,7 +52,7 @@ jobs:
 
       - name: Build the website
         run: |
-          yarn install --immutable
+          yarn install  --immutable
```
Comment (willdurand) [inline .github/workflows/gcp_deploy_prod.yaml:55]:
```suggestion
          yarn install --immutable
```

File: .github/workflows/gcp_deploy.yaml [hunk 5]
```diff
@@ -11,14 +11,16 @@ jobs:
...
+          fetch-depth: 0
 
       - name: Setup Node.js
         uses: actions/setup-node@v3
         with:
           node-version: "18"
 
       - name: Install dependencies
-        run: yarn install --immutable
+        run: yarn install
```
Comment (willdurand) [inline .github/workflows/gcp_deploy.yaml:23]:
```suggestion
        run: yarn install --immutable
```

File: .github/workflows/gcp_deploy_prod.yaml (hunk 3 above)
Comment (pattersonbl2) [reply .github/workflows/gcp_deploy_prod.yaml:6]: Long as the same tag is being pushed and no one pushes anything not following the pattern

**Comment by pattersonbl2** [comment]
@willdurand  i have commit all the suggestion.  if it's ok i would like to merge this in today so i can start planning the move to prod.

**Review by willdurand — APPROVED** [review]

**Pull request merged**